src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -63,15 +63,16 @@ object Parsers {
       atPos(Position(start, end, point))(t)
 
     def atPos[T <: Positioned](start: Offset, point: Offset)(t: T): T =
-      atPos(start, point, in.lastOffset max start)(t)
+      if (in.lastOffset > start) atPos(start, point, in.lastOffset)(t) else t
```

Comment:
No, that's actually a common case. It always happens when the generated tree does not consume any source. In that case we don't set a position and wait for the position to be set by the enclosing node. I'll add a comment to explain.